kv/kvserver: TestMergeQueue failed

kv/kvserver.TestMergeQueue [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3125207&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3125207&tab=artifacts#/) on master @ [52f87b5a495e33704b82b93729bd61fb76d5fecc](https://github.com/cockroachdb/cockroach/commits/52f87b5a495e33704b82b93729bd61fb76d5fecc):


```
=== RUN   TestMergeQueue
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestMergeQueue045735340
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestMergeQueue
    client_merge_test.go:4402: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestMergeQueue045735340
--- FAIL: TestMergeQueue (7.97s)
=== RUN   TestMergeQueue/sticky-bit-expiration
    client_merge_test.go:4395: ranges unexpectedly merged
    --- FAIL: TestMergeQueue/sticky-bit-expiration (0.54s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestMergeQueue PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/kv nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMergeQueue.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -4422,14 +4422,19 @@ func TestMergeQueue(t *testing.T) {
 	})
 
 	t.Run("sticky-bit-expiration", func(t *testing.T) {
-		skip.WithIssue(t, 66942, "flakey test")
-		manualSplitTTL := time.Millisecond * 200
+		manualSplitTTL := 10 * time.Millisecond
 		reset(t)
 		store.MustForceMergeScanAndProcess()
 		verifyUnmerged(t, store, lhsStartKey, rhsStartKey)
 
+		// Pause the clock for the duration of this subtest so that
+		// the sticky bit does not expire prematurely.
+		manualClock.Pause()
+		defer manualClock.Resume()
+
 		// Perform manual merge and verify that no merge occurred.
-		split(t, rhsStartKey.AsRawKey(), tc.Servers[0].Clock().Now().Add(manualSplitTTL.Nanoseconds(), 0) /* expirationTime */)
+		exp := tc.Servers[0].Clock().Now().Add(manualSplitTTL.Nanoseconds(), 0)
+		split(t, rhsStartKey.AsRawKey(), exp /* expirationTime */)
 		clearRange(t, lhsStartKey, rhsEndKey)
 		store.MustForceMergeScanAndProcess()
 		verifyUnmerged(t, store, lhsStartKey, rhsStartKey)

--- a/pkg/util/hlc/hlc.go
+++ b/pkg/util/hlc/hlc.go
@@ -172,6 +172,14 @@ func (m *HybridManualClock) Pause() {
 	m.mu.Unlock()
 }
 
+// Resume resumes the hybrid manual clock; the passage of time continues to
+// cause clock to tick.
+func (m *HybridManualClock) Resume() {
+	m.mu.Lock()
+	m.mu.nanosAtPause = 0
+	m.mu.Unlock()
+}
+
 // UnixNano returns the local machine's physical nanosecond
 // unix epoch timestamp as a convenience to create a HLC via
 // c := hlc.NewClock(hlc.UnixNano, ...).

--- a/pkg/util/hlc/hlc_test.go
+++ b/pkg/util/hlc/hlc_test.go
@@ -373,9 +373,10 @@ func TestHybridManualClock(t *testing.T) {
 	require.LessOrEqual(t, c.Now().WallTime, UnixNano())
 	require.LessOrEqual(t, UnixNano(), c.Now().WallTime)
 
-	m.Increment(10)
-	require.LessOrEqual(t, c.Now().WallTime, UnixNano()+10)
-	require.LessOrEqual(t, UnixNano()+10, c.Now().WallTime)
+	inc := time.Second.Nanoseconds()
+	m.Increment(inc)
+	require.LessOrEqual(t, c.Now().WallTime, UnixNano()+inc)
+	require.LessOrEqual(t, UnixNano()+inc, c.Now().WallTime)
 }
 
 // TestHybridManualClockPause test the Pause() functionality of the
@@ -391,8 +392,14 @@ func TestHybridManualClockPause(t *testing.T) {
 	require.Equal(t, now, c.Now().WallTime)
 	time.Sleep(10 * time.Millisecond)
 	require.Equal(t, now, c.Now().WallTime)
-	m.Increment(10)
-	require.Equal(t, now+10, c.Now().WallTime)
+	inc := time.Second.Nanoseconds()
+	m.Increment(inc)
+	require.Equal(t, now+inc, c.Now().WallTime)
+	m.Resume()
+	trueNow := UnixNano()
+	require.LessOrEqual(t, trueNow+inc, c.Now().WallTime)
+	time.Sleep(10 * time.Millisecond)
+	require.Less(t, trueNow+inc, c.Now().WallTime)
 }
 
 func TestHLCMonotonicityCheck(t *testing.T) {

